packages/infra/lib/api-stack.ts

```diff
@@ -281,9 +281,6 @@ export class APIStack extends Stack {
     const cdaToVisualizationLambda = this.setupCdaToVisualization({
       lambdaLayers,
       vpc: this.vpc,
-      bucketName: isSandbox(props.config)
-        ? props.config.sandboxSeedDataBucketName
-        : medicalDocumentsBucket.bucketName,
```

Comment:
But for the visualization im only going to be referencing the bucket that gets passed so why would i need the buckets?

Reply:
it needs access to them so it can read/write files, right? so we'll have to give access to both buckets since the lambda can access either one of them

Reply:
Oh gotcha yea they both have access to both buckets were just not passing the bucketName through the env rather through params

Reply:
Yeah, that's what I was saying... we're giving access to the buckets the lambda uses outside of the function that creates the lambda. This is not encapsulated - ideally we pass the resources the lambda needs to its creation function and it does what it needs (pass env vars, allow access, etc.).

I know this was like this already, but I'm trying to make this stack code easier to be refactored in the future, it won't scale well the way it is.

NABD if you don't want to move it now though - we can be more strict when we build those in the first place (although the encapsulated pattern has been around for quite some time now)

Reply:
Ah gotcha i see now i moved it for those lambdas